### PR TITLE
Dashboard widget

### PR DESCRIPTION
**Pull request by bwmarkle**

### CODE REVIEW

File: src/Library/Configs.php [hunk 1]
```diff
@@ -103,6 +105,58 @@ public static function get( $key = null ) {
...
+	 * return "boldgrid-backup".
+	 *
+	 * @since 2.9.0
+	 *
+	 * @return string
+	 */
+	public static function getFileSlug() {
+		$file = explode( '/', self::get('file') );
+
+		return $file[0];
```
Comment (cssjoe) [inline src/Library/Configs.php]: Is that always going to have input to explode?  I rather use `return reset( $file );` to get the first index or FALSE.

File: src/Library/Dashboard/SortWidgets.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+		$hasChangedLibrary = ! empty( $hasChangedLibrary );
+
+		switch( $userChangedOrder && ! $hasChangedLibrary ) {
+			case true:
+				$this->sortCustomOrder();
+				break;
+			case false:
+				$this->sortGlobal();
+				break;
+		}
```
Comment (cssjoe) [inline src/Library/Dashboard/SortWidgets.php]: Why not `$userChangedOrder && ! $hasChangedLibrary ? $this->sortCustomOrder() : $this->sortGlobal();` ?

File: src/Library/NewsWidget.php [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	public function __construct() {
+		Filter::add( $this );
+	}
+
+	/**
+	 * Enqueue scripts.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param string $hook
```
Comment (cssjoe) [inline src/Library/NewsWidget.php]: Missing @params description.

File: src/Library/Dashboard/SortWidgets.php (hunk 2 above)
Comment (bwmarkle) [reply src/Library/Dashboard/SortWidgets.php]: image

File: src/Library/NewsWidget.php [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+?>			<li title="<?php echo esc_attr( $post->excerpt->rendered ); ?>">
+				<?php if ( ! empty( $featuredImage ) ) {
+					echo '<img class="bglib-featured" src="' . esc_url( $featuredImage ) . '" />';
+				} ?>
+				<p class="bglib-date">
+					<?php echo date( 'M jS, Y', strtotime( $post->date ) ); ?>
+				</p>
+				<p class="bglib-title">
+					<a href='<?php echo $post->link ?>' target='_blank'><?php echo $post->title->rendered; ?></a>
```
Comment (cssjoe) [inline src/Library/NewsWidget.php]: Missing semicolon.

File: src/Library/Plugin/Plugin.php [hunk 5]
```diff
@@ -20,31 +20,147 @@
  * @since 2.7.7
  */
 class Plugin {
+	/**
+	 * An array of children plugins.
+	 *
+	 * Elements of this array are of this class, Plugin (recursion).
+	 *
+	 * @since 2.10.0
```
Comment (cssjoe) [inline src/Library/Plugin/Plugin.php]: This @since should be "2.9.0".

File: src/Library/Plugin/Plugin.php [hunk 6]
```diff
@@ -20,31 +20,147 @@
...
 
 	/**
 	 * Plugin slug.
 	 *
 	 * Examples: boldgrid-backup, boldgrid-backup-premium, etc.
 	 *
-	 * @var  string
+	 * @var string
 	 * @since 2.7.7
+	 * @access private
```
Comment (cssjoe) [inline src/Library/Plugin/Plugin.php]: Change to "protected".

File: src/Library/Plugin/Plugin.php [hunk 7]
```diff
@@ -54,4 +170,142 @@ public function getDownloadUrl() {
...
+
+	/**
+	 * Get a plugin's slug from its file.
+	 *
+	 * For example, if the plugin's file is:
+	 * boldgrid-backup-premium/boldgrid-backup-premium.php
+	 * The plugin's slug is:
+	 * boldgrid-backup-premium
+	 *
+	 * @since 2.10.0
```
Comment (cssjoe) [inline src/Library/Plugin/Plugin.php]: This `@since` should be "2.9.0".

File: src/Library/Plugin/Plugin.php [hunk 8]
```diff
@@ -54,4 +170,142 @@ public function getDownloadUrl() {
...
+	 * @return string
+	 */
+	public function getSlug() {
+		return $this->slug;
+	}
+
+	/**
+	 * Set our child plugins.
+	 *
+	 * @since 2.10.0
```
Comment (cssjoe) [inline src/Library/Plugin/Plugin.php]: This `@since` should be "2.9.0".

File: src/Library/Dashboard/SortWidgets.php (hunk 2 above)
Comment (bwmarkle) [reply src/Library/Dashboard/SortWidgets.php]: updated.

File: src/Library/NewsWidget.php (hunk 3 above)
Comment (bwmarkle) [reply src/Library/NewsWidget.php]: Fixed. https://github.com/BoldGrid/library/commit/429c5c6412fc39d4d172db21ea01902f6e99f902

**Pull request merged**